aips/aip-116.md

```diff
@@ -265,7 +305,38 @@ To ensure the safety of our users, we want to take a conservative approach when
 
 - A wallet MUST NOT allow the approval of signing requests that do not match on the `domain` field.
 - The `domain` and `uri` field is expected to be read from a trusted data source such as the browser window. The `address`, `version`, and `chainId` fields are expected to be read from the wallet. **These fields are expected to be bound during the construction of the SIWA message if not present**.
-- Wallet implementors MUST verify the following fields to prevent phishing attacks: `domain` , `address` , `uri` , `version` , and `chainId` . If the bound fields do not match the fields presented in the signing request, errors must be displayed to the users detailing them out. It is suggested to add additional steps to the approval process, such as an opt-in checkmark field, that confirms the user’s intent if errors are present.
+- Wallet implementors MUST ensure that the bound fields (`address`, `chainId`, `domain`, `uri`, and `version`) match their corresponding fields in the `AptosSignInInput` _if present_. If the bound fields do not match the fields presented in the signing request, errors must be displayed to the users detailing the issues.
+- In the case that the signing Aptos account was rotated, the bound `address` field MUST be the originating address of the authentication key.
+
+### Wallet Interactions
+
+- Prior to signing, the wallet MAY consult users for preferences, such as the selection of one `address` out of many or the `network` to use.
+
+#### Resource Injection
+
+Wallets may inject additional information into the `resources` field for **wallet specified resources** and in any **wallet specified formats**.
```

Comment:
I think the reader should be reminded when & where this injection happens: is it when the wallet gets the `AptosSignInInput` message and fills it in?